tools/autograd/gen_variable_type.py

```diff
@@ -463,25 +560,36 @@ def emit_dispatch(i, option):
         env = {}
 
         args = []
-        formal_args = ['Tensor & self']
         python_params = args_without_self(option['python_arguments'])
+        has_self = python_params != option['python_arguments']
```

Comment:
Uh that seems like something that'll easily break in the future, especially that we still haven't clarified what an option even is. Can't we hold a boolean flag in the option?

Reply:
done.